triton-vm/src/program.rs

```diff
@@ -385,6 +385,109 @@ impl ProfileLine {
     }
 }
 
+#[derive(Clone, Debug, BFieldCodec)]
+pub struct PublicInput {
+    pub stream: Vec<BFieldElement>,
+}
+
+impl From<Vec<BFieldElement>> for PublicInput {
+    fn from(stream: Vec<BFieldElement>) -> Self {
+        PublicInput { stream }
+    }
+}
+
+impl From<&Vec<BFieldElement>> for PublicInput {
+    fn from(stream: &Vec<BFieldElement>) -> Self {
+        PublicInput {
+            stream: stream.to_owned(),
+        }
+    }
+}
+
+impl From<&[BFieldElement]> for PublicInput {
+    fn from(stream: &[BFieldElement]) -> Self {
+        PublicInput {
+            stream: stream.to_vec(),
+        }
+    }
+}
+
+impl From<Vec<u64>> for PublicInput {
+    fn from(stream: Vec<u64>) -> Self {
+        PublicInput {
+            stream: stream.iter().map(|&element| element.into()).collect(),
+        }
+    }
+}
+
+impl From<[u64; 0]> for PublicInput {
+    fn from(_stream: [u64; 0]) -> Self {
+        PublicInput { stream: vec![] }
+    }
+}
+
+impl PublicInput {
+    pub fn new(stream: Vec<BFieldElement>) -> Self {
+        PublicInput { stream }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SecretInput {
+    pub stream: Vec<BFieldElement>,
```

Comment:
I don't really like the name "stream" because it suggests some kind of logspace computation on a large amount of linearly arrayed data, along with the capacity to compose processes and chain streams. But I'm not sure what alternative to use. Source, perhaps?

Reply:
Also, see my earlier comment about splitting up this field into two, one for each type of divination.